src/components/ReferralButton.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState } from 'react';
+import {
+  Button, Dropdown, Card, Input, Space, Tooltip,
+} from 'antd';
+import { ShareAltOutlined, DownOutlined } from '@ant-design/icons';
+import { Auth } from 'aws-amplify';
+import endUserMessages from '../utils/endUserMessages';
+import pushNotificationMessage from '../utils/pushNotificationMessage';
+
+const { TextArea } = Input;
+
+const initialMessage = 'Hi,\n\nCheck out Cellscope from Biomage. It will make your single-cell analysis easier.';
+
+// Valid email regex based on RFC2822 - https://regexr.com/2rhq7
+const emailRegex = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
+
+const ReferralButton = () => {
+  const [visible, setVisible] = useState(false);
+  const [email, setEmail] = useState('');
+  const [isEmailValid, setIsEmailValid] = useState(false);
+  const [customMessage, setCustomMessage] = useState(initialMessage);
+
+  const HOOK_URL = 'aHR0cHM6Ly9ob29rcy5zbGFjay5jb20vc2VydmljZXMvVDAxNTVEWkZWTTAvQjAyQVk0ODQxQ0cvQ0x3Mms4dTBtMkUzcDBVNUhhbjBqeTBv'; // pragma: allowlist secret
```

Comment:
Do you know what this HOOK_URL is refering to?

Reply:
it's the 64-byte encoded form of ReferralBot's [webhook URL](https://api.slack.com/messaging/webhooks). It's encoded to provide more security